### PR TITLE
`./x doc library --open` opens `std`

### LINKED ISSUE #105898: Documenting private items of standard library with stage 1 compiler fails
Running `RUSTDOCFLAGS=--document-private-items ./x doc --stage 1 library` fails with this error:

```
 Documenting alloc v0.0.0 (/home/lukas/code/rust/library/alloc)
error[E0080]: evaluation of constant value failed
    --> library/alloc/src/collections/btree/node.rs:1679:38
     |
1679 |         const TRAVERSAL_PERMIT: () = panic!();
     |                                      ^^^^^^^^ the evaluated program panicked at 'explicit panic', library/alloc/src/collections/btree/node.rs:1679:38
     |
     = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not document `alloc`

Caused by:
  process didn't exit successfully: `/home/lukas/code/rust/build/bootstrap/debug/rustdoc --edition=2021 --crate-type lib --crate-name alloc library/alloc/src/lib.rs --target x86_64-unknown-linux-gnu -o /home/lukas/code/rust/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/doc --cfg 'feature="compiler-builtins-c"' -Zunstable-options --check-cfg 'names()' --check-cfg 'values()' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -Z unstable-options --resource-suffix 1.68.0 --markdown-css rust.css --markdown-no-toc --index-page /home/lukas/code/rust/src/doc/index.md -C metadata=d6cfa37a8de9f6b1 -L dependency=/home/lukas/code/rust/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/release/deps -L dependency=/home/lukas/code/rust/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc/release/deps --extern compiler_builtins=/home/lukas/code/rust/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/release/deps/libcompiler_builtins-adcb81061c726f2b.rmeta --extern core=/home/lukas/code/rust/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/release/deps/libcore-29b0da95ca4e4f3b.rmeta -Csymbol-mangling-version=legacy -Zunstable-options -Zunstable-options '--check-cfg=values(bootstrap)' '--check-cfg=values(stdarch_intel_sde)' '--check-cfg=values(no_fp_fmt_parse)' '--check-cfg=values(no_global_oom_handling)' '--check-cfg=values(no_rc)' '--check-cfg=values(no_sync)' '--check-cfg=values(freebsd12)' '--check-cfg=values(backtrace_in_libstd)' '--check-cfg=values(target_env,"libnx")' '--check-cfg=values(target_os,"watchos")' '--check-cfg=values(target_arch,"asmjs","spirv","nvptx","nvptx64","le32","xtensa")' '--check-cfg=values(dont_compile_me)' '--check-cfg=values(rustix_use_libc)' --document-private-items -Dwarnings '-Wrustdoc::invalid_codeblock_attributes' --crate-version 1.68.0-dev '-Zcrate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/")'` (exit status: 1)
```

Without `--stage 1` it emits the same error, but "passes". Found in https://github.com/rust-lang/rust/pull/105848, specifically https://github.com/rust-lang/rust/pull/105848/commits/3223b6c4ca2b4743010e42c5e1e50896ef8f10ae.

Also, not really related, but `./x doc --open library` doesn't open anything.

@rustbot label C-bug
@rustbot claim

### PR DESCRIPTION
**Pull request by lukas-code**
fix https://github.com/rust-lang/rust/issues/105898

### CODE REVIEW

File: library/alloc/src/collections/btree/node.rs [hunk 1]
```diff
@@ -1003,7 +1006,10 @@ impl<BorrowType: marker::BorrowType, K, V>
     /// `edge.descend().ascend().unwrap()` and `node.ascend().unwrap().descend()` should
     /// both, upon success, do nothing.
     pub fn descend(self) -> NodeRef<BorrowType, K, V, marker::LeafOrInternal> {
-        let _ = BorrowType::TRAVERSAL_PERMIT;
+        const {
+            assert!(BorrowType::TRAVERSAL_PERMIT);
+        }; // #[cfg(bootstrap)] remove ;
```
Comment (jyn514) [inline library/alloc/src/collections/btree/node.rs:1011]: why should this be removed in the next cycle?

**Review by jyn514 — COMMENTED** [review]
The bootstrap changes LGTM, but I had a question about the changes to btree.

File: library/alloc/src/collections/btree/node.rs (hunk 1 above)
Comment (lukas-code) [reply library/alloc/src/collections/btree/node.rs:1011]: Inline const parses as expression with block after https://github.com/rust-lang/rust/pull/105142

Comment (lukas-code) [reply library/alloc/src/collections/btree/node.rs:1011]: Oh, apparently this is in beta already and just blocked on https://github.com/rust-lang/rust/pull/105741

Comment (jyn514) [reply library/alloc/src/collections/btree/node.rs:1011]: Ok, let's wait until that gets merged so you can rebase over it and avoid needing cfg(bootstrap).

**Review by thomcc — APPROVED** [review]
r=me on these changes, but lets wait for https://github.com/rust-lang/rust/pull/105741 to land

**Comment by lukas-code** [comment]
The libs change got merged in https://github.com/rust-lang/rust/pull/105741, so this is just the `./x doc --open` fix now.

@rustbot label -T-libs -S-blocked +S-waiting-on-review

**Pull request merged**